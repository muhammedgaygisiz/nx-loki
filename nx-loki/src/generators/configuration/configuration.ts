import { formatFiles, generateFiles, Tree } from "@nx/devkit";
import * as path from "path";
import { ConfigurationGeneratorSchema } from "./schema";

export async function configurationGenerator(
  tree: Tree,
  options: ConfigurationGeneratorSchema
) {
  const projectRoot = `apps/${options.name}`;

  generateFiles(tree, path.join(__dirname, "files"), projectRoot, { tmpl: "" });

  await formatFiles(tree);
}

export default configurationGenerator;
