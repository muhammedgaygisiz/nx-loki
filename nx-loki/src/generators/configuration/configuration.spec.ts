import { createTreeWithEmptyWorkspace } from "@nx/devkit/testing";
import { Tree, readProjectConfiguration } from "@nx/devkit";

import { configurationGenerator } from "./configuration";
import { ConfigurationGeneratorSchema } from "./schema";

describe("configuration generator", () => {
  let tree: Tree;
  const options: ConfigurationGeneratorSchema = { name: "test" };

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
  });

  it("should run successfully", async () => {
    await configurationGenerator(tree, options);

    const LOKI_CONFIG_FILE = `apps/${options.name}/loki.config.js`;
    expect(tree.exists(LOKI_CONFIG_FILE)).toBeTruthy();

    const PROJECT_PACKAGE_JSON = `apps/${options.name}/package.json`;
    expect(tree.exists(PROJECT_PACKAGE_JSON)).toBeTruthy();

    const lokiConfigFile = tree.read(LOKI_CONFIG_FILE, "utf-8");
    expect(lokiConfigFile).toMatchSnapshot();

    const packageJsonFile = JSON.parse(
      tree.read(PROJECT_PACKAGE_JSON, "utf-8")
    );
    expect(packageJsonFile).toMatchSnapshot();
  });
});
