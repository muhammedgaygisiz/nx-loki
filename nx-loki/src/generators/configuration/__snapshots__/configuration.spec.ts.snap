// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configuration generator should run successfully 1`] = `
"module.exports = {
  configurations: {
    'chrome.laptop': {
      target: 'chrome.docker',
      width: 1366,
      height: 768,
      deviceScaleFactor: 1,
      mobile: false,
    },
    'chrome.iphone7': {
      target: 'chrome.docker',
      preset: 'iPhone 7',
    },
  },
};
"
`;

exports[`configuration generator should run successfully 2`] = `
{
  "description": "Dummy package.json since loki itself expects its.",
  "name": "test-loki",
}
`;
